test/transformer.base.js

```diff
@@ -611,6 +611,47 @@ module.exports = function base(transformer, pathname, transformer_name) {
         var socket = new Socket(server.addr);
       });
 
+      it('does not introduce data loss for async connection events', function (done) {
+        var pre;
+
+        primus.on('connection', function (spark, next) {
+          setTimeout(function () {
+            pre = 'async';
+
+            socket.write('in');
+
+            next();
```

Comment:
This should be a sync call, which triggers the flusing of the `data` event. So the `order` should be processed normally.